python/fledge/services/core/api/plugins/remove.py

```diff
@@ -0,0 +1,165 @@
+                                                                        plugin_type == 'filter'):
+            get_enabled_plugin = await get_enabled_or_disabled_status(e['service'])
+            _logger.info(get_enabled_plugin)
+            if plugin_name in [x['plugin'] for x in enabled_asset_list['rows']] or e['plugin'] in filter_used:
+                list_of_enabled_plugin.append(e)
+        _logger.info(list_of_enabled_plugin)
+    return list_of_enabled_plugin
+
+
+async def get_enabled_or_disabled_status(service_name: str):
+    """
+        check plugin state(enabled or disabled)
+    """
+    storage_client = connect.get_storage_async()
+    payload_data = PayloadBuilder().SELECT('id', 'enabled').WHERE(['schedule_name', '=', service_name]).payload()
+    enabled_service_list = await storage_client.query_tbl_with_payload('schedules', payload_data)
+    return enabled_service_list['rows']
+
+
+# TODO: notification service state(enabled or disabled)
+async def check_notification_service_state(plugin_type: str, plugin_name: str):
+    pass
+
+
+def purge_plugin(plugin_type: str, name: str):
+    """
+        Remove plugin based on platform
+    """
+    _logger.info("Plugin removal started...")
+    name = name.replace('_', '-').lower()
+    plugin_name = 'fledge-{}-{}'.format(plugin_type, name)
+    stdout_file_path = common.create_log_file(action='delete', plugin_name=plugin_name)
+    get_platform = platform.platform()
+    try:
+        package_manager = 'yum' if 'centos' in get_platform or 'redhat' in get_platform else 'apt'
+        if package_manager == 'yum':
+            cmd = "sudo {} -y remove {} > {} 2>&1".format(package_manager, plugin_name, stdout_file_path)
+        else:
+            cmd = "sudo {} -y purge {} > {} 2>&1".format(package_manager, plugin_name, stdout_file_path)
+        code = os.system(cmd)
+        installed_plugin = PluginDiscovery.get_plugins_installed(plugin_type, False)
+        if name in [plugin['name'] for plugin in installed_plugin] or code != 0:
+            raise KeyError("Plugin is not installed by package manager.".format(name))
+    except KeyError as ex:
+        try:
+            _logger.info(_FLEDGE_ROOT)
+            path = PYTHON_PLUGIN_PATH+'{}/{}'.format(plugin_type, name)
+            if os.path.isdir(path):
+                for root, dirs, files in os.walk(path):
+                    if '__pycache__' in dirs:
+                        os.system('rm -rf {}'.format(os.path.join(root, '__pycache__')))
+                    else:
+                        for file in files:
+                            os.remove(os.path.join(root, file))
+                        for name in dirs:
+                            os.rmdir(os.path.join(root, name))
+                code = os.system('rm -rf {}'.format(path))
+            else:
+                path = C_PLUGINS_PATH + '{}/{}'.format(plugin_type, name)
```

Comment:
@rajeshkumar426

There is an issue here:

I have south plugin Random:

$FOGLAMP_ROOT/plugins/south/Random/libRandom.so

It is not removed.

I added a log message and I see:

____ removing C plugin /home/ubuntu/source/develop/FogLAMP/plugins/south/random

This does nothing and no error is logged.

First you need to check the error and then you need to keep the original name:

at the beginning of purge_plugin I see
name = name.replace('_', '-').lower()

You should add before that
originalName = name
name = name.replace('_', '-').lower()

Then when the plugin is not managed by package manager you can remove it by using originalName

Reply:
Resolved